app/controllers/students_controller.rb

```diff
@@ -1,9 +1,58 @@
 class StudentsController < ApplicationController
     before_action :authenticate_user!
     before_action :set_student, only: %i[ show edit update destroy ]
-    # GET /students
+    # GET /student
     def index
-      @students = Student.where(teacher: current_user.email)
+		@students = Student.where(teacher: current_user.email)
+		@tags = Set[]
+		@emails = Set[]
+
+		@courses_taken = Hash[]
+		@semesters_taken = Hash[]
+		for student in @students do
+			@tags.add(student.tags)
+
+			# Figure out each student's course/semester they have taken
+			@courses_taken[student.course_id] = Course.find(student.course_id).course_name
+			@semesters_taken[student.course_id] = Course.find(student.course_id).semester
+
+		end
+		@semesters = Set[]
+		@sections = Set[]
+		@course_names = Set[]
+        @course_ids = Array[]
+		for record in Course.all do
+			@semesters.add(record.semester)
+			@sections.add(record.section)
+			@course_names.add(record.course_name)
+            @course_ids.append(record.id)
+		end
+
+        if request.post?
```

Comment:
we should do a check like this, since it's always supposed to be a GET, you check whether any of the params is not nil instead

Reply:
Fixed! Thanks for the suggestion, this is a much better approach